lib/IDE/CodeCompletion.cpp

```diff
@@ -2204,7 +2204,18 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
         SemanticContextKind::None,
         expectedTypeContext);
     Builder.setAssociatedDecl(MD);
-    Builder.addBaseName(MD->getNameStr());
+    auto moduleName = MD->getName();
+ 
+    // This checks if module aliasing was used. For example, when editing
+    // `import ...`, and `-module-alias Foo=Bar` was passed, we want to show
+    // Foo as an option to import, instead of Bar (name of the binary), as
+    // Foo is the name that should appear in source files.
+    auto aliasedName = Ctx.getRealModuleName(moduleName, ASTContext::ModuleAliasLookupOption::aliasFromRealName);
```

Comment:
IIUC, `MD->getName()` _should_ return the aliased name in the first place. I guess this is needed because the names returned from `ASTContext::getVisibleTopLevelModuleNames()` are "real" names, right? Is it possible to modify it or (`*ModuleLoader::collectVisibleTopLevelModuleNames()`) to return aliased names?

Reply:
Correct, the names here are the "real" names (binary names) that were searched through the search paths.  Moving the logic to return aliases to those functions doesn't seem ideal as they are lower-level functions that should not have know about the input flag -module-alias.

Reply:
Per discussion, this is to be addressed in a separate PR if needed after merging this. cc @rintaro